File: ruby/nth-prime/nth_prime.rb
class Prime
  def self.nth(nth_prime)
    raise ArgumentError if nth_prime == 0
    index           = 1
    nth_prime_found = 2
    number          = 3
    while index != nth_prime
      if is_prime(number)
        index += 1
        nth_prime_found = number
      end
      number += 1
    end
    nth_prime_found
  end

  def self.is_prime(num)
    (2..Math.sqrt(num).to_i).each { |n| return false if num % n == 0 }
    true
  end
end

File: elixir/nth-prime/nth_prime.exs
defmodule Prime do

  def nth(goal), do: get_nth_prime(2, 1, goal)

  defp is_prime(n) when n <= 3, do: true

  defp is_prime(n) do
    last = :math.sqrt(n) |> Float.floor |> round
    not (2..last |> Enum.any?(&(rem(n, &1) == 0)))
  end

  defp get_nth_prime(_, _, goal) when goal < 1, do: raise ArgumentError

  defp get_nth_prime(num, goal, goal), do: num

  defp get_nth_prime(num, count, goal) do
    new_count = if is_prime(num + 1), do: count + 1, else: count
    get_nth_prime(num + 1, new_count, goal)
  end

end
